board/safety.h

```diff
@@ -643,10 +631,25 @@ bool steer_angle_cmd_checks(int desired_angle, bool steer_control_enabled, const
   }
   desired_angle_last = desired_angle;
 
-  // Angle should be the same as current angle while not steering
-  violation |= (!steer_control_enabled &&
-                  ((desired_angle < (angle_meas.min - 1)) ||
-                  (desired_angle > (angle_meas.max + 1))));
+  // check that commanded angle value isn't too far from measured, used to limit torque for some safety modes
+  if (controls_allowed && steer_control_enabled && limits.enforce_angle_error) {
+    if (vehicle_speed > limits.angle_error_limit_speed) {
+      // val must always be near angle_meas, limited to the maximum value
+      // add 1 to not false trigger the violation
+      int highest_allowed = CLAMP(angle_meas.max + limits.max_angle_error + 1, -limits.max_steer, limits.max_steer);
+      int lowest_allowed = CLAMP(angle_meas.min - limits.max_angle_error - 1, -limits.max_steer, limits.max_steer);
+
+      // check for violation
+      violation |= max_limit_check(desired_angle, highest_allowed, lowest_allowed);
+    }
+  }
+
+  // Angle should either be 0 or same as current angle while not steering
+  if (!steer_control_enabled) {
+    violation |= (limits.inactive_angle_is_zero ? (desired_angle != 0) :
+                    ((desired_angle < (angle_meas.min - 1)) ||
+                    (desired_angle > (angle_meas.max + 1))));
+  }
```

Comment:
seems like a cppcheck bug (`(desired_angle != false) :` passes), this should be compliant 😞. not fixed on latest master either